API/Controllers/UsersController.cs

```diff
@@ -1,37 +1,103 @@
 ﻿using API.Models;
+using API.Models.AccountModels;
 using API.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using System.Threading.Tasks;
 
 namespace API.Controllers
 {
     [Route("api/v1/[controller]/[action]")]
     [Produces(ApiContentType)]
     [ApiController]
+    [Authorize]
     public class UsersController : ApiControllerBase
     {
-        public UsersController(ApiContext context, Iserializer serializer, IErrorFormatter errorFormatter) :
-            base(context, serializer, errorFormatter) { }
+        private readonly UserManager<UserDTO> _userManager;
+        private readonly SignInManager<UserDTO> _signInManager;
+        private readonly JwtService _jwt;
+        public UsersController(
+            ApiContext context,
+            Iserializer serializer,
+            IErrorFormatter errorFormatter,
+            IConfiguration config,
+            UserManager<UserDTO> userManager,
+            SignInManager<UserDTO> signInManager) :
+            base(context, serializer, errorFormatter)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+            _jwt = new JwtService(config);
+        }
 
         [HttpPost]
-        public IActionResult Signup(UserDTO user)
+        [AllowAnonymous]
+        public async Task<IActionResult> Signup(RegisterModel model)
         {
             if (!IsValidApiRequest())
             {
                 return ApiBadRequest("Invalid headers!");
             }
 
-            return NotImplementedError("Signup Not Implemented!");
+            if (ModelState.IsValid)
+            {
+                var user = new UserDTO { UserName = model.Username ?? model.Email, Email = model.Email };
+                var result = await _userManager.CreateAsync(user, model.Password);
+                if (result.Succeeded)
+                {
+                    return Created("", new
+                    {
+                        token = _jwt.GenerateSecurityToken(new User()
+                        {
+                            Username = user.UserName,
+                            Email = user.Email
+                        })
+                    });
+                }
+
+                return BadRequest(result);
+            }
+
+            return BadRequest(ModelState);
```

Comment:
This error here as doesn't conform to the specified protocol. Have you tested this on Postman?
```json
{
    "errors": {
        "Password": [
            "The Password field is required."
        ]
    },
    "type": "https://tools.ietf.org/html/rfc7231#section-6.5.1",
    "title": "One or more validation errors occurred.",
    "status": 400,
    "traceId": "|8a339a34-4dd69c1bb625fc03."
}
```

Reply:
Fixed.